### PR TITLE
NO MERGE: Multi Load Vehicle

### PR DESCRIPTION
**Pull request by ViljoenFritz**
Summary of this change:

Changed Schema: load_vehicles are now referenced form loads

TODO: Add tests for TruckArrival service. 
TODO: Test and fix RMD functionality

CHANGELOG entry:
### Added
- .
### Changed
- [Changed Schema: load_vehicles are now referenced form loads](https://github.com/NoSoft-SA/nspack/pull/407/commits/a0159524ce450f82c27ecbee6f7ce393aded301d)
### Fixed
- .

### CODE REVIEW

**Review by jcsjcs — COMMENTED** [review]
Like the idea of inheriting Dry::Structs. Never bothered to test that one before.
I only got about 3/4 down the page. Have to stop for now, so I'm leaving these comments...

File: lib/base_interactor.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class BaseInteractor
   # @param client_settings [Hash]
   # @param context [Hash]
   # @param logger [Hash]
-  def initialize(user, client_settings, context, logger)
+  def initialize(user, client_settings = {}, context = {}, logger = {})
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/base_interactor.rb:12]: cool

File: lib/finished_goods/services/truck_arrival.rb [hunk 2]
```diff
@@ -64,26 +68,21 @@ def create_container
       return unless @container_id.nil?
 
       container_id = container_repo.create_load_container(@container_params)
-      repo.log_status(:load_containers, container_id, 'CREATED', user_name: @user_name)
+      log_status(:load_containers, container_id, "CREATED FOR LOAD #{@load_id}")
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/finished_goods/services/truck_arrival.rb:71]: i like it

File: lib/finished_goods/interactors/load_container_interactor.rb [hunk 3]
```diff
@@ -20,17 +20,10 @@ def create_load_container(params)
...
       res = validate_load_container_params(params)
       return validation_failed_response(res) if res.failure?
 
-      # test for changes
-      instance = load_container(id).to_h.reject { |k, _| %i[id active].include?(k) }
-      return success_response("Load: #{instance[:load_id]}", instance) if instance == res.output
-
-      # update date field if weight changed
-      res.output[:verified_gross_weight_date] = Time.now if instance[:verified_gross_weight] != res.output[:verified_gross_weight]
-
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/finished_goods/interactors/load_container_interactor.rb:33]: thanks, it was a unnecessary check

**Pull request closed without merging**